cmd/crank/beta/trace/internal/resource/xrm/client.go

```diff
@@ -88,6 +111,29 @@ func (kc *Client) GetResourceTree(ctx context.Context, root *resource.Resource)
 	return root, nil
 }
 
+// getKnownResourceChildren returns children for known resources.
+func getKnownResourceChildren(res *resource.Resource) (children []*resource.Resource) {
+	gvk := res.Unstructured.GroupVersionKind()
+	switch {
+	// provider-kubernetes Objects, we want to show the manifest as a child
+	case gvk.GroupKind() == schema.GroupKind{Group: "kubernetes.crossplane.io", Kind: "Object"}:
```

Comment:
Do we really want to start hard coding MR types into our CLI?

What other MRs might we add here apart from provider-kubernetes Objects?

Reply:
I agree it's not ideal but I think that `Objects` are a unique case in that they encapsulate other resources.  I could also see provider-helm `Releases` added here to expose the name of the chart being applied.  It would be nice to have a more generic implementation that could be expanded as needed but I'm not sure there is a need beyond these two cases?

Reply:
I needed it, so I wrote it. I don't see any drawback to start this way and see how it goes, if we see a lot of PRs adding support for providers (I highly doubt we will), we can reconsider the choice and define some standard providers should adhere to instead.

Reply:
How would we define what is acceptable to include here as a known resource?

I'm less convinced we won't see folks wanting to teach the CLI about other known resources once this door is opened. Once we have the coupling, it will mean needing to update the CLI to be aware of MR schema changes.

If we feel strongly that this only applies to `provider-kubernetes` and `provider-helm`, and we can clearly frame the rationale behind making the CLI aware of those two providers and no others I feel okay with moving forward. Otherwise, I'm concerned about the coupling. AFAIK this is the only place in the whole c/c codebase where we're adding hardcoded knowledge of MRs.

Reply:
Reading https://github.com/crossplane/crossplane/issues/5188#issuecomment-1874254366, is there a desire to see the whole object? Or do we just want to know its name?

Is that stored in the `crossplane.io/external-name` annotation? Do we currently surface that in trace?

Reply:
Actually the gvk, name and namespace would be enough to have it shown as an additional children. It's just a nice side effect that you get the unwrapped resource also in the json output setting the whole thing as children instead of just having it shown properly.

For example an `Object` with a `uid` as name doesn't say much, but seeing it's actually a `ConfigMap` named `foo` in the namespace `bar` helps understanding what's going on.

About external resources, no, we don't show that right now, but I could see adding support for additional columns based on labels/annotations similar to what kubectl does, so no need to handle it in any special way.